jobs/base_model_validator_test.py

```diff
@@ -0,0 +1,225 @@
+    [models.NAMES.base_model, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MockModel(base_models.BaseModel):
+    pass
+
+
+class BaseValidatorDoFnTests(unittest.TestCase):
+    def setUp(self):
+        self.base_validator_fn = base_model_validator.BaseValidatorDoFn()
+
+    def test_clone(self):
+        model = base_models.BaseModel(id='123', deleted=True)
+        clone = self.base_validator_fn.clone(model)
+
+        self.assertEqual(model.id, clone.id)
+        self.assertEqual(model, clone)
+        self.assertIsNot(model, clone)
+        self.assertIsInstance(clone, base_models.BaseModel)
+
+    def test_clone_with_changes(self):
+        model = base_models.BaseModel(id='123', deleted=True)
+        clone = self.base_validator_fn.clone(model, deleted=False)
+
+        self.assertNotEqual(model, clone)
+        self.assertIsNot(model, clone)
+        self.assertIsInstance(clone, base_models.BaseModel)
+        self.assertTrue(model.deleted)
+        self.assertFalse(clone.deleted)
+
+    def test_clone_sub_class(self):
+        class DerivedModel(base_models.BaseModel):
+            """Simple model with an extra 'field' string property."""
+
+            field = datastore_services.StringProperty()
+
+        model = DerivedModel(id='123', field='original')
+        clone = self.base_validator_fn.clone(model)
+
+        self.assertEqual(model, clone)
+        self.assertIsNot(model, clone)
+        self.assertIsInstance(clone, DerivedModel)
+        self.assertEqual(model.field, 'original')
+        self.assertEqual(clone.field, 'original')
+
+    def test_clone_sub_class_with_changes(self):
+        class DerivedModel(base_models.BaseModel):
+            """Simple model with an extra 'field' string property."""
+
+            field = datastore_services.StringProperty()
+
+        model = DerivedModel(id='123', field='original')
+        clone = self.base_validator_fn.clone(model, field='updated')
+
+        self.assertNotEqual(model, clone)
+        self.assertIsNot(model, clone)
+        self.assertIsInstance(clone, DerivedModel)
+        self.assertEqual(model.field, 'original')
+        self.assertEqual(clone.field, 'updated')
+
+
+class BaseModelValidatorTests(unittest.TestCase):
+
+    def setUp(self):
+        self.now = datetime.datetime.utcnow()
+        self.year_ago = (
+            datetime.datetime.utcnow() - datetime.timedelta(weeks=52))
+        self.year_later = (
+            datetime.datetime.utcnow() + datetime.timedelta(weeks=52))
+
+    def test_validate_deleted_reports_error_for_old_deleted_model(self):
+
+        with TestPipeline(runner=DirectRunner()) as p:
+            test_model = MockModel(
+                id='123',
+                deleted=True,
+                created_on=self.year_ago,
+                last_updated=self.year_ago
+            )
+            pcoll = p | beam.Create([test_model])
+
+            output = (pcoll | beam.ParDo(
```

Comment:
Let's make the codebase convention for beam: start multi-line pipes (`|`) on new lines. We can worry about adding lint checks later, but for this PR let's just follow it manually.
```suggestion
            output = (pcoll
                | beam.ParDo(
```